src/control/connector.rs

```diff
@@ -252,3 +258,53 @@ impl From<State> for u32 {
         }
     }
 }
+
+/// Subpixel order of the connected sink
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum SubPixel {
+    /// Unknown geometry
+    Unknown,
+    /// Horizontal RGB
+    HorizontalRgb,
+    /// Horizontal BGR
+    HorizontalBgr,
+    /// Vertical RGB
+    VerticalRgb,
+    /// Vertical BGR
+    VerticalBgr,
+    /// No geometry
+    None,
+}
+
+impl From<u32> for SubPixel {
+    fn from(n: u32) -> Self {
+        // These values are not defined in drm_mode.h
+        // They were copied from linux's drm_connector.h
+        // Don't mistake those for ones used in xf86DrmMode.h (libdrm offsets those by 1)
+        match n {
+            0 => Self::Unknown,
+            1 => Self::HorizontalRgb,
+            2 => Self::HorizontalBgr,
+            3 => Self::VerticalRgb,
+            4 => Self::VerticalBgr,
+            5 => Self::None,
+            _ => Self::Unknown,
+        }
+    }
+}
+
+impl From<SubPixel> for u32 {
+    fn from(state: SubPixel) -> Self {
+        // These values are not defined in drm_mode.h
+        // They were copied from linux's drm_connector.h
+        // Don't mistake those for ones used in xf86DrmMode.h (libdrm offsets those by 1)
+        match state {
+            SubPixel::Unknown => 0,
+            SubPixel::HorizontalRgb => 1,
+            SubPixel::HorizontalBgr => 2,
+            SubPixel::VerticalRgb => 3,
+            SubPixel::VerticalBgr => 4,
+            SubPixel::None => 5,
+        }
+    }
+}
```

Comment:
Given we are not using the same definition as `xf86DrmMode`, I am wondering if we should expose this conversion trait at all. The returned values don't have any particular meaning to downstream and there is no place to publically document why these are seemingly shifted by 1.

Reply:
If we use `#[repr(u32)]` for this, downstream can simply cast it as `u32` without needing this trait.

On the other hand, maybe these values could vary between different kernels? If that's the case, we would probably want these values to be put into `drm-ffi` and pull them in here.

Reply:
Removed all together, I agree with Drakulix, I believe that using values different than xf86DrmMode would only confuse users.